File: python-project-euler-100/p024.py
from math import trunc
from Solution import Solution

class P024(Solution):

    # Factoradic method
    def string_permutation(self, n, str):
   
        s = []
        result = ""
        str = list(str)
        n = n - 1
    
        for i in range(1, len(str) + 1):
            s.append(n % i)
            n = int(n / i)
    
        for i in range(len(str)):
            a = s[-1]
            result += str[a]

            for j in range(a, len(str) - 1):
                str[j] = str[j + 1]

            str[j + 1] = '\0'
            s.pop()

        return result

    def solve(self):
        self.problem_number = 24
        n = 1000000
        data = '0123456789'
        result = int(self.string_permutation(n, data))

        return result

def main():
    P024().run()

if __name__ == "__main__":
    main()

File: java-project-euler-100/src/main/java/net/talaatharb/projecteuler100/P024.java
package net.talaatharb.projecteuler100;

import java.util.Stack;

/**
 * P024
 * 
 * @author mharb
 *
 */
public class P024 extends Solution {

	/**
	 * Problem solution entry point
	 * 
	 * @param args
	 */
	public static void main(String[] args) {
		new P024().run();
	}

	// Factoradic method
	private String stringPermutation(int n, String str) {

		Stack<Integer> s = new Stack<Integer>();
		StringBuilder result = new StringBuilder();

		n = n - 1;

		for (int i = 1; i < str.length() + 1; i++) {
			s.add(n % i);
			n = n / i;
		}

		for (int i = 0; i < str.length(); i++) {
			int a = s.peek();
			result.append(str.charAt(a));
			int j;

			for (j = a; j < str.length() - 1; j++)
				str = str.substring(0, j) + str.charAt(j + 1) + str.substring(j + 1);

			str = str.substring(0, j + 1);
			s.pop();
		}
		return result.toString();
	}

	@Override
	public Number solve() {
		problemNumber = 24;
		final int n = 1000000;
		final String data = "0123456789";
		
		return Long.parseLong(stringPermutation(n, data));
	}

}
